File: JJSwiftClass_02/03-block回调的回顾/03-block回调的回顾/ViewController.m
//
//  ViewController.m
//  03-block回调的回顾
//
//  Created by xiaomage on 15/11/29.
//  Copyright © 2015年 xiaomage. All rights reserved.
//

#import "ViewController.h"
#import "HttpTool.h"

@interface ViewController ()

/** 网络请求的对象 */
@property (nonatomic, strong) HttpTool *httpTool;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.httpTool = [[HttpTool alloc] init];
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    __weak ViewController *weakSelf = self;
    
    [self.httpTool requestData:^{
        NSLog(@"获取到数据");
        NSLog(@"更新界面:%@", [NSThread currentThread]);
        
        weakSelf.view.backgroundColor = [UIColor purpleColor];
    }];
}

- (void)dealloc
{
    NSLog(@"ViewController----dealloc");
}

@end

File: JJSwiftClass_02/05-Swift中的懒加载/05-Swift中的懒加载/ViewController.swift
//
//  ViewController.swift
//  05-Swift中的懒加载
//
//  Created by xiaomage on 15/11/29.
//  Copyright © 2015年 xiaomage. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    // 1>数据用到时再加载
    // 2>数据只会加载
    lazy var names : [String] = {
        
        // () -> [String] in
        
        print("加载数据")
        return ["why", "yz", "lmj", "lnj"]
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
        print(names)
        print(names)
        print(names)
    }

}

